src/misc/lv_event.c

```diff
@@ -86,27 +87,29 @@ lv_result_t lv_event_send(lv_event_list_t * list, lv_event_t * e, bool preproces
 void lv_event_add(lv_event_list_t * list, lv_event_cb_t cb, lv_event_code_t filter,
                   void * user_data)
 {
-    list->cnt++;
-    list->dsc = lv_realloc(list->dsc, list->cnt * sizeof(lv_event_dsc_t));
-    LV_ASSERT_MALLOC(list->dsc);
-    if(list->dsc == NULL) return;
+    if(list->size == 0) {
+        lv_array_init(list, 1, sizeof(lv_event_dsc_t));
+    }
+
+    lv_event_dsc_t dsc = { 0 };
+    dsc.cb = cb;
+    dsc.filter = filter;
+    dsc.user_data = user_data;
 
-    list->dsc[list->cnt - 1].cb = cb;
-    list->dsc[list->cnt - 1].filter = filter;
-    list->dsc[list->cnt - 1].user_data = user_data;
+    lv_array_resize(list, lv_array_size(list) + 1);
```

Comment:
At the moment there is a possible double memory allocation issue:
if `list` is empty:
1. then it is initialized to capacity 1 in line 91: `lv_array_init(list, 1, sizeof(lv_event_dsc_t));`
2. then here list is resized to 2: `lv_array_resize(list, lv_array_size(list) + 1);`

So, I think lv_array_push_back() function should increase the array automatically, and so if list is initialized with capacity 1 (and size=0), then there will be no double memory allocation.
https://github.com/lvgl/lvgl/pull/5300#discussion_r1451475640

Reply:
Added `else` to eliminate the extra dsc memory.